fili-core/src/main/java/com/yahoo/bard/webservice/async/jobs/stores/HashJobStore.java

```diff
@@ -51,4 +61,43 @@ public HashJobStore() {
     public Observable<JobRow> getAllRows() {
         return Observable.from(store.values());
     }
+
+    @Override
+    public Observable<JobRow> getFilteredRows(Set<ApiJobStoreFilter> apiJobStoreFilters)
+            throws IllegalArgumentException {
+        return Observable.from(
+                store.entrySet().stream()
+                        .filter(entry -> satisfiesFilters(apiJobStoreFilters, entry.getValue()))
+                        .map(entry -> entry.getValue())
+                        .collect(Collectors.toSet())
+        );
+    }
+
+    /**
+     * This method checks if the given JobRow satisfies all the ApiJobStoreFilters and returns true if it does.
+     * If a JobField in any of the filters is not a part the JobRow, this method throws a IllegalArgumentException.
+     *
+     * @param apiJobStoreFilters  A Set of ApiJobStoreFilters specifying the different conditions to be satisfied
+     * @param jobRow  The JobRow which needs to be inspected
+     *
+     * @return true if the JobRow satisfies all the filters, false otherwise
+     *
+     * @throws IllegalArgumentException if a JobField in any of the filters is not a part the JobRow
+     */
+    private boolean satisfiesFilters(Set<ApiJobStoreFilter> apiJobStoreFilters, JobRow jobRow)
+            throws IllegalArgumentException {
+        for (ApiJobStoreFilter filter : apiJobStoreFilters) {
+            JobField jobField = filter.getJobField();
+            if (!jobRow.containsKey(jobField)) {
+                Set<JobField> jobFields = jobRow.keySet();
+                LOG.debug(JOBFIELD_NOT_PRESENT_IN_JOB_META_DATA.logFormat(jobField, jobFields));
+                throw new IllegalArgumentException(JOBFIELD_NOT_PRESENT_IN_JOB_META_DATA.format(jobField, jobFields));
+            }
+
+            if (!filter.getValues().contains(jobRow.get(jobField))) {
+                return false;
+            }
+        }
+        return true;
```

Comment:
You know, if you really wanted to, you could pull the body of the for loop into another method `satisfiesFilter` that only looks at one JobRow and one Filter. Then, you can write this function as a very simple stream:

```java
return apiJobStoreFilters.stream().allMatch(filter -> satisfiesFilter(jobRow, filter);
```

And the method `satisfiesFilter` can be simplified to:

```java
 if(!jobRow.containsKey(jobField)) {
   // exception stuff
}
return filter.getValues().contains(jobRow.get(jobField));
```

In fact, we'll probably want to do this anyway, because we should also support the other filter operations, and this method will become too complex for my tastes if it handles the checks for every possible filter operation.